doc/whats_new/v1.0.rst

```diff
@@ -138,6 +138,14 @@ Changelog
   :pr:`17743` by :user:`Maria Telenczuk <maikia>` and
   :user:`Alexandre Gramfort <agramfort>`.
 
+:mod:`sklearn.manifold`
+.......................
+
+- |Fix| The numerical precision in `manifold._utils._binary_search_perplexity`
+  was changed from `float` to `double` to prevent underflow issues
+  during affinity matrix computation for t-SNE.
+  :pr:`19472` by :user:`Dmitry Kobak <dkobak>`.
```

Comment:
I would give less details about the private function, and link to the estimator doc page.
```suggestion
- |Fix| Change numerical precision to prevent underflow issues
  during affinity matrix computation for :class:`TSNE`.
  :pr:`19472` by :user:`Dmitry Kobak <dkobak>`.
```

Since this fix might change results for some users, I would also add an entry in the `Changed models` section at the top of this file. Since it is the first entry there (other fixes probably should have been listed there), you can get an example of the typical formatting in `doc/whats_new/v0.24.rst`.

Reply:
Commited your suggestion and added an entry to the `Changed models` section.